client/src/redux/actions/onboarding.js

```diff
@@ -122,3 +132,27 @@ export const fetchSubscribedNumber = userSession => async dispatch => {
   dispatch({ type: FETCH_SUBSCRIBED_NUMBER, phoneNumber });
   return phoneNumber;
 };
+
+export const setTempUnit = (userSession, currentObservations, currentUnit, nextUnit) => async dispatch => {
+  if (nextUnit === currentUnit) return;
+  const nextObservations = currentObservations.map(obs => {
+    return {
+      ...obs,
+      physical: {
+        ...obs.physical,
+        feverSeverity: convertUnits(nextUnit, obs.physical.feverSeverity),
+      },
+    };
+  });
+  dispatch(changedObservations(nextObservations));
+  dispatch({ type: SET_TEMP_UNIT, nextUnit });
+  userSession.putFile('tempUnit', nextUnit);
```

Comment:
Something to note: files saved without `.json` at the end will not be deleted with the "Delete All Data" button